### PR TITLE
CSM-6197: Auto-integration Fix

### PR DESCRIPTION
**Pull request by mellis-uptycs**
AWS Lambda poll's the sqs queue so it grabs more than one message and in the "event" there are sometimes more than one record so this iterates through the records instead of just integrating the role for the first message.

The git diff looks weird so the only changes I made where to line 8 where I do a check to make sure event isn't empty.
I then added the for loop on line 13.  As well I made a refactor of [0] for [i]

### CODE REVIEW

**Review by radis-uptycs — APPROVED** [review]

File: lambdas/roleManager.js [hunk 1]
```diff
@@ -5,70 +5,78 @@ const utils = require('./utils');
...
-        callback(new Error('Invalid RequestType. Valid values: "Create" or "Delete"'));
-        return;
-    }
-
-    // Check if all required attributes are set
-    if (!req.IntegrationName || !req.AccountIds || !req.UptAccountId || !req.ExternalId || !req.RequestType) {
-        callback(new Error(`Request:${JSON.stringify(req, null, 2)} is not valid. Required fields are missing`));
-        return;
-    }
+    for (let i = 0; i < event.Records.length; i++) {
```
Comment (gbonthu) [inline lambdas/roleManager.js:13]: Please replace with for-each loop

File: lambdas/roleManager.js [hunk 2]
```diff
@@ -5,70 +5,78 @@ const utils = require('./utils');
...
-        callback(new Error('AccountIds must be an array of strings with length of 1'));
-        return;
-    }
-
-    const stsCreds = await utils.stsCreds(req.AccountIds[0]);
-    const iamClient = await utils.iamClient(stsCreds, 'aws-global');
-    const sqsClient = await utils.sqsClient(event.Records[0].awsRegion);
+        if (!event.Records[i] || !event.Records[i].body) {
+            callback(new Error('No event specified'));
+            return;
```
Comment (gbonthu) [inline lambdas/roleManager.js:17]: be careful with returns.
returning on Error is fine here after the callback.
Please review other places

File: lambdas/roleManager.js [hunk 3]
```diff
@@ -5,70 +5,78 @@ const utils = require('./utils');
...
-    try {
-        if (req.RequestType === 'Delete') {
-            await utils.detachPoliciesFromRole(iamClient, req.IntegrationName);
-            await utils.deleteIntegrationRole(iamClient, req.IntegrationName);
-            console.log(`Successfully deleted integration role ${req.IntegrationName} from account ${req.AccountIds[0]}`);
-            await utils.sendResponse(sqsClient, event.Records[0].eventSourceARN, req.IntegrationName, response);
+        // Check if all required attributes are set
+        if (!req.IntegrationName || !req.AccountIds || !req.UptAccountId || !req.ExternalId || !req.RequestType) {
+            callback(new Error(`Request:${JSON.stringify(req, null, 2)} is not valid. Required fields are missing`));
             return;
```
Comment (gbonthu) [inline lambdas/roleManager.js:30]: cannot return!

File: lambdas/roleManager.js [hunk 4]
```diff
@@ -5,70 +5,78 @@ const utils = require('./utils');
...
+            // Check if we have to replace existing role
+            if (req.Force && roleExists) {
+                await utils.detachPoliciesFromRole(iamClient, req.IntegrationName);
+                await utils.deleteIntegrationRole(iamClient, req.IntegrationName);
+            } else if (roleExists) {
+                const msg = `Role ${req.IntegrationName} already exists`;
+                response.message = msg;
+                console.error(msg);
+                await utils.sendResponse(sqsClient, event.Records[i].eventSourceARN, req.IntegrationName, response);
+                return;
```
Comment (gbonthu) [inline lambdas/roleManager.js:70]: cannot return. Continue?

**Review by gbonthu — APPROVED** [review]

**Pull request merged**